sdk/eventhub/Azure.Messaging.EventHubs/tests/EventHubClient/EventHubClientLiveTests.cs

```diff
@@ -215,5 +216,188 @@ await using (var client = new EventHubClient(connectionString))
+
+                await using (var client = new EventHubClient(connectionString, new EventHubClientOptions { TransportType = TransportType.AmqpWebSockets }))
+                {
+                    var partition = (await client.GetPartitionIdsAsync()).First();
+
+                    await using (var consumer = client.CreateConsumer(partition, EventPosition.Latest))
+                    {
+                        Assert.That(async () => await consumer.ReceiveAsync(1, TimeSpan.Zero), Throws.Nothing);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="EventHubClient" /> is able to
+        ///   connect to the Event Hubs service.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        public async Task ClientCannotRetrieveMetadataWhenClosed(bool sync)
+        {
+            await using (var scope = await EventHubScope.CreateAsync(1))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                await using (var client = new EventHubClient(connectionString))
+                {
+                    var partition = (await client.GetPartitionIdsAsync()).First();
+
+                    Assert.That(async () => await client.GetPropertiesAsync(), Throws.Nothing);
+                    Assert.That(async () => await client.GetPartitionPropertiesAsync(partition), Throws.Nothing);
+
+                    if (sync)
+                    {
+                        client.Close();
+                    }
+                    else
+                    {
+                        await client.CloseAsync();
+                    }
+
+                    Assert.ThrowsAsync<OperationCanceledException>(async () => await client.GetPartitionIdsAsync());
+                    Assert.ThrowsAsync<ObjectDisposedException>(async () => await client.GetPropertiesAsync());
+                    Assert.ThrowsAsync<ObjectDisposedException>(async () => await client.GetPartitionPropertiesAsync(partition));
+                }
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="EventHubClient" /> is able to
+        ///   connect to the Event Hubs service.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ClientCannotRetrievePartitionPropertiesWhenPartitionIdIsInvalid()
+        {
+            await using (var scope = await EventHubScope.CreateAsync(1))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                await using (var client = new EventHubClient(connectionString))
+                {
+                    // Some invalid partition values. These will fail on the service side.
+                    var invalidPartitions = new[]
+                    {
+                        "XYZ",
```

Comment:
Can we please drive these from `[TestCase]`.    Generally speaking, the test shouldn't work against different cases in the same run;  makes it far more difficult to understand exactly which failed and why.

Reply:
Done.